openoa/utils/plot.py

```diff
@@ -940,8 +313,21 @@ def plot_by_id(
         id_col(:obj:`str`): The asset_id column (or index column) in `df`.
         x_axis(:obj:`str`): Independent variable to plot, should align with a column in `df`.
         y_axis(:obj:`str`): Dependent variable to plot, should align with a column in `df`.
-        return_fig(:obj:`bool`): Indicator for if the figure and axes objects should be returned,
-            by default False.
+        max_cols(:obj:`int`, optional): The maximum number of columns in the plot. Defaults to 4.
+        xlim(:obj:`tuple[float, float]`, optional): A tuple of the x-axis (min, max) values.
+            Defaults to (None, None).
+        ylim(:obj:`tuple[float, float]`, optional): A tuple of the y-axis (min, max) values.
+            Defaults to (None, None).
+        xlabel(:obj:`str` | None): The x-axis label, if None, then :py:attr:`x_axis` will be used.
+            Defaults to None.
+        ylabel(:obj:`str` | None): The y-axis label, if None, then :py:attr:`x_axis` will be used.
+            Defaults to None.
+        return_fig(:obj:`bool`, optional): Set to True to return the figure and axes objects,
+            otherwise set to False. Defaults to False.
+        plot_kwargs(:obj:`dict`, optional): Additional keyword arguments that should be passed
+            to `ax.scatter`. Defaults to {}.
+        legend_kwargs(:obj:`dict`, optional): Additional keyword arguments that should be passed to
+            `ax.legend`. Defaults to {}.
```

Comment:
based on the list of arguments in the function definition, `legend_kwargs` should be changed to `figure_kwargs` and listed before `plot_kwargs`

Reply:
Thanks for flagging this! It looks like I deleted the wrong docstring item when copying over the base information from another function.